spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -217,30 +217,26 @@ contract RolloverVault is
         uint256 amount;
     }
 
-    /// @notice Burns notes and sends a proportional share vault's assets back to {msg.sender}.
+    /// @notice Burns notes and sends a proportional share of vault's assets back to {msg.sender}.
     /// @param notes The amount of notes to be burnt.
     /// @return The list of asset tokens and amounts redeemed.
     function redeem(uint256 notes) external nonReentrant whenNotPaused returns (TokenAmount[] memory) {
         uint256 totalNotes = totalSupply();
-        uint256 deployedCount_ = deployedCount();
-        uint256 earnedCount_ = earnedCount();
-        uint256 assetCount = 1 + deployedCount_ + earnedCount_;
-
-        TokenAmount[] memory redemptions = new TokenAmount[](assetCount);
+        uint256 deployedCount_ = _deployed.length();
+        uint256 assetCount = 2 + deployedCount_;
```

Comment:
I could see a class of bugs open up if `underlying` and `earned` are the same token.

I don't see this being the case for this deployment, so perhaps the easiest way to handle it is to add an assert statement in initialization somewhere that they're not equal.

It might work as is... but it's weird to have an asset appear twice in a list of assets, or the number of assets not equal the length of the list of assets, etc. It's a case I'd rather not have to test for or think about.

Reply:
We can assert `underlying != earned` in the initializer